### PR TITLE
Add package publishing to the release workflow

### PR DESCRIPTION
**Pull request by alex-f-k**

### CODE REVIEW

File: .github/workflows/build-and-publish-sdk.yml [hunk 1]
```diff
@@ -17,30 +18,50 @@ jobs:
           distribution: 'corretto'
           java-version: '11'
```
Comment (ASitnikov) [inline .github/workflows/build-and-publish-sdk.yml:19]: We use Java 8

File: .github/workflows/build-and-publish-sdk.yml [hunk 2]
```diff
@@ -17,30 +18,50 @@ jobs:
...
+        run: ./gradlew :payments-workflows:build -x testFixturesJar -Pversion="${GITHUB_REF_NAME:1}" --no-daemon
+
+      - name: 'Publish to maven'
+        env:
+          GPG_PASSPHRASE: ${{ secrets.GPG_PASSPHRASE }}
+          MAVEN_USERNAME: ${{ secrets.MAVEN_USERNAME }}
+          MAVEN_PASSWORD: ${{ secrets.MAVEN_PASSWORD }}
+        run: ./gradlew publishAllPublicationsToOSSRHRepository -Pversion="${GITHUB_REF_NAME:1}" -Psigning.gnupg.keyName=D3838CD92688873C -Psigning.gnupg.passphrase="$GPG_PASSPHRASE" --no-daemon
+
+      - name: 'Publish to Github Packages'
```
Comment (ASitnikov) [inline .github/workflows/build-and-publish-sdk.yml:37]: Is it still needed?

File: build.gradle.kts [hunk 3]
```diff
@@ -28,3 +30,10 @@ tasks {
         }
     }
 }
+
+tasks.dokkaHtmlMultiModule.configure {
+    pluginsMapConfiguration.put(
+        "org.jetbrains.dokka.base.DokkaBase",
+        """ { "footerMessage": "Copyright ${LocalDate.now().year} Exactpro Systems Limited" } """,
```
Comment (ASitnikov) [inline build.gradle.kts:37]:
```suggestion
        """ { "footerMessage": "Copyright 2022-${LocalDate.now().year} Exactpro Systems Limited" } """,
```

I'd use utc date

File: payments-contracts/build.gradle.kts [hunk 4]
```diff
@@ -15,3 +20,70 @@ cordapp {
         versionId(1)
     }
 }
+
+publishing {
```
Comment (ASitnikov) [inline payments-contracts/build.gradle.kts:24]: Add conventions plugin for ossrh publishing

File: payments-contracts/build.gradle.kts [hunk 5]
```diff
@@ -15,3 +20,70 @@ cordapp {
...
+                    }
+                }
+            }
+        }
+    }
+    repositories {
+        maven {
+            name = "test"
+            setUrl("file://" + System.getenv("HOME") + "/.m2/repository/")
+        }
```
Comment (ASitnikov) [inline payments-contracts/build.gradle.kts:59]: Is it still needed?

File: build.gradle.kts (hunk 3 above)
Comment (ASitnikov) [reply build.gradle.kts:37]: Add an Redmine task to move this code to the conventions plugin

**Review by ASitnikov — CHANGES_REQUESTED** [review]

File: .github/workflows/build-and-publish-sdk.yml [hunk 6]
```diff
@@ -17,30 +18,50 @@ jobs:
...
           cd workflows/build/libs/
-          ls | xargs -I {} sh -c 'sha256sum "{}" > {}.sha256'
+          ls *.jar | xargs -I {} sh -c 'sha256sum "{}" > {}.sha256'
+
+      - name: 'Release'
+        uses: softprops/action-gh-release@v1
+        with:
+          files: |
+            contracts/build/libs/*
+            workflows/build/libs/*
```
Comment (ASitnikov) [inline .github/workflows/build-and-publish-sdk.yml:57]: Change paths to `payments-xxx`

File: build.gradle.kts (hunk 3 above)
Comment (ASitnikov) [reply build.gradle.kts]: > Add an Redmine task to move this code to the conventions plugin

Or TODO in code

Comment (alex-f-k) [reply build.gradle.kts]: Created issue RM#82866

File: build-logic/dsl-plugins/src/main/kotlin/com.gevamu.publish-cordapp-conventions.gradle.kts [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+                scm {
+                    connection.set("scm:git:git://github.com/gevamu/corda-payments-sdk.git")
+                    developerConnection.set("scm:git:ssh://github.com:gevamu/corda-payments-sdk.git")
+                    url.set("https://github.com/gevamu/corda-payments-sdk/tree/master")
+                }
+                developers {
+                    developer {
+                        name.set("Gevamu")
+                        organization.set("Gevamu")
+                        organizationUrl.set("https://github.com/gevamu")
```
Comment (ASitnikov) [inline build-logic/dsl-plugins/src/main/kotlin/com.gevamu.publish-cordapp-conventions.gradle.kts:34]: https://gevamu.com

File: build-logic/dsl-plugins/src/main/kotlin/com.gevamu.publish-cordapp-conventions.gradle.kts [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+                }
+                scm {
+                    connection.set("scm:git:git://github.com/gevamu/corda-payments-sdk.git")
+                    developerConnection.set("scm:git:ssh://github.com:gevamu/corda-payments-sdk.git")
+                    url.set("https://github.com/gevamu/corda-payments-sdk/tree/master")
+                }
+                developers {
+                    developer {
+                        name.set("Gevamu")
+                        organization.set("Gevamu")
```
Comment (ASitnikov) [inline build-logic/dsl-plugins/src/main/kotlin/com.gevamu.publish-cordapp-conventions.gradle.kts:33]: Exactpro Systems Limited

File: build-logic/dsl-plugins/src/main/kotlin/com.gevamu.publish-cordapp-conventions.gradle.kts [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+                licenses {
+                    license {
+                        name.set("The Apache License, Version 2.0")
+                        url.set("https://www.apache.org/licenses/LICENSE-2.0.txt")
+                    }
+                }
+                scm {
+                    connection.set("scm:git:git://github.com/gevamu/corda-payments-sdk.git")
+                    developerConnection.set("scm:git:ssh://github.com:gevamu/corda-payments-sdk.git")
+                    url.set("https://github.com/gevamu/corda-payments-sdk/tree/master")
```
Comment (ASitnikov) [inline build-logic/dsl-plugins/src/main/kotlin/com.gevamu.publish-cordapp-conventions.gradle.kts:28]: https://github.com/gevamu/corda-payments-sdk

File: build.gradle.kts [hunk 10]
```diff
@@ -28,3 +31,11 @@ tasks {
         }
     }
 }
+
+tasks.dokkaHtmlMultiModule.configure {
+    val currentYear = LocalDate.now(ZoneId.of("UTC")).year
+    pluginsMapConfiguration.put(
+        "org.jetbrains.dokka.base.DokkaBase",
+        """ { "footerMessage": "Copyright ${if (2022 == currentYear) currentYear else "2022-$currentYear"} Exactpro Systems Limited" } """,
```
Comment (ASitnikov) [inline build.gradle.kts:39]: Remove checks for 2022. These checks will work only today and tomorrow. It doesn't make sense to add this code

File: payments-workflows/build.gradle.kts [hunk 11]
```diff
@@ -16,9 +21,23 @@ dependencies {
...
     contract {
-        name("Payment Workflows")
+        name("Payments Workflows")
         vendor("Exactpro Systems LLC")
         licence("Apache License, Version 2.0")
         versionId(1)
     }
 }
+
+tasks.withType<Javadoc> {
```
Comment (ASitnikov) [inline payments-workflows/build.gradle.kts:31]: Is it still needed? I think I disabled javadoc tasks in kotlin conventions plugin

**Review by ASitnikov — CHANGES_REQUESTED** [review]

**Review by ASitnikov — APPROVED** [review]
Looks good to me

**Pull request merged**